apps/admin/src/resources/signing-keys/list.tsx

```
@@ -166,18 +416,58 @@ function RotateSigningKeyButton() {
           </DialogHeader>
-          <p className="text-sm">
-            Create a new signing key and schedule revocation of the current key.
-            New tokens will be signed with the new key immediately; existing
-            keys remain valid during a 24-hour grace period.
+          {isSaml ? (
+            <p className="text-sm">
+              Create a new key pair and publish its certificate in the SAML
+              metadata straight away, but keep signing with the current one
+              until it activates. Send the new certificate to every service
+              provider during that window — they cannot fetch it themselves, and
+              assertions signed by a certificate they don't trust are rejected.
```

ESLint

[error] `'` can be escaped with `&apos;`, `&lsquo;`, `&`#39`;`, `&rsquo;`.

(react/no-unescaped-entities)